File: typescript/rna-transcription/rna-transcription.ts
export function toRna(dna: string) {
  const dnaToRna = new Map([
    ["G", "C"],
    ["C", "G"],
    ["T", "A"],
    ["A", "U"],
  ]);

  if (dna.match(/[^GCTA]/)) throw new Error("Invalid input DNA.");
  // spread operator to convert string to array of characters, map over each character, and join back into a string
  return [...dna].map((nucleotide) => dnaToRna.get(nucleotide)).join("");
}

File: csharp/rna-transcription/RnaTranscription.cs
using System.Linq;

public static class RnaTranscription
{
    // G -> C, C -> G, T -> A, A -> U
    public static string ToRna(string nucleotide) =>
        new string(nucleotide
            .Select(c => c == 'G' ? 'C' : c == 'C' ? 'G' : c == 'T' ? 'A' : c == 'A' ? 'U' : c).ToArray());
}